read.go

```
@@ -269,10 +266,52 @@ func readNativeFrames(d dicomio.Reader, parsedData *Dataset, fc chan<- *frame.Fr
 
 	debug.Logf("readNativeFrames:\nRows: %d\nCols:%d\nFrames::%d\nBitsAlloc:%d\nSamplesPerPixel:%d", MustGetInts(rows.Value)[0], MustGetInts(cols.Value)[0], nFrames, bitsAllocated, samplesPerPixel)
 
+	bytesAllocated := bitsAllocated / 8
+	bytesRead = bytesAllocated * samplesPerPixel * pixelsPerFrame * nFrames
+	if bitsAllocated == 1 {
+		bytesRead = pixelsPerFrame * samplesPerPixel / 8 * nFrames
+	}
+
+	skipOne := false
+	switch {
+	case uint32(bytesRead) == vl: // we good nothing
+	case uint32(bytesRead) == vl-1 && vl%2 == 0:
+		skipOne = true
+	case uint32(bytesRead) == vl-1 && vl%2 != 0:
+		return nil, 0, fmt.Errorf("vl=%d: %w", vl, ErrorExpectedEvenLength)
+	default:
+		if !parsedData.opts.allowMismatchPixelDataLength {
+			return nil, 0, fmt.Errorf("expected_vl=%d actual_vl=%d %w", bytesRead, vl, ErrorMismatchPixelDataLength)
+		}
+		data := make([]byte, vl)
+		_, err = io.ReadFull(d, data)
+		if err != nil {
+			panic(err)
```

why panic when we can return an error?